### PR TITLE
ProcessHandler: Make timeout of `stream_select()` configurable

### PR DESCRIPTION
**Pull request by ericemmrich**
In our application, any request has always been blocked for 1 second and made it significantly slower. Rather than needing to overwrite almost the entire class with a custom implementation, it would be nice to have this timeout configurable.

### CODE REVIEW

**Comment by ericemmrich** [comment]
@Seldaek What would be required to move this forward? :)

**Comment by Seldaek** [comment]
Do you use ProcessHandler in prod? What for? Just out of curiosity sorry..

Anyway the PR seems fine so I'll merge but I'll need some more free time before I can triage other stuff and tag a release.

File: src/Monolog/Handler/ProcessHandler.php [hunk 1]
```diff
@@ -43,6 +43,11 @@ class ProcessHandler extends AbstractProcessingHandler
      */
     private array $pipes = [];
 
+    /**
+     * @var int
+     */
+    private $timeout;
```
Comment (Seldaek) [inline src/Monolog/Handler/ProcessHandler.php:49]:
```suggestion
    private float $timeout;    
```

**Comment by ericemmrich** [comment]
> Do you use ProcessHandler in prod? What for? Just out of curiosity sorry..
> 
> Anyway the PR seems fine so I'll merge but I'll need some more free time before I can triage other stuff and tag a release.

No worries, thanks for picking this up! :)

We use the ProcessHandler for redirecting some logs to the `stderr` of process ID 1, so that they are displayed properly in the container logs:

```
            'handler' => \Monolog\Handler\ProcessHandler::class,
            'with' => [
                'command' => 'cat >> /proc/1/fd/2',
            ],
```

**Review by Seldaek — COMMENTED** [review]
Just a suggestion to allow passing a float as timeout so you can set microseconds too, because just having 0 or 1 is not allowing for a lot of precision.

File: src/Monolog/Handler/ProcessHandler.php [hunk 2]
```diff
@@ -146,7 +153,7 @@ protected function selectErrorStream()
         $empty = [];
         $errorPipes = [$this->pipes[2]];
 
-        return stream_select($errorPipes, $empty, $empty, 1);
+        return stream_select($errorPipes, $empty, $empty, $this->timeout);
```
[the remark this answers is not in the document]
Comment (ericemmrich) [reply src/Monolog/Handler/ProcessHandler.php:156]: That's smart :)

**Comment by Seldaek** [comment]
Alright, thanks

**Pull request merged**